ocaml/xapi/xapi_pgpu.ml

```diff
@@ -307,3 +354,25 @@ let disable_dom0_access ~__context ~self =
   if not (Pool_features.is_enabled ~__context Features.Integrated_GPU)
   then raise Api_errors.(Server_error (feature_restricted, []));
   update_dom0_access ~__context ~self ~action:`disable
+
+(* This must be run LOCALLY on the host that is about to start a VM that is
+ * going to use a vgpu backed by an AMD MxGPU pgpu. *)
+let mxgpu_vf_setup ~__context =
+  let localhost = Helpers.get_localhost ~__context in
+  (* From the modprobe(8) manpage:
+   * --first-time
+   *     Normally, modprobe will succeed (and do nothing) if told to insert a
+   *     module which is already present or to remove a module which isn't
+   *     present. This is ideal for simple scripts; however, more complicated
+   *     scripts often want to know whether modprobe really did something: this
+   *     option makes modprobe fail in the case that it actually didn't do
+   *     anything. *)
+  ignore (Forkhelpers.execute_command_get_output !Xapi_globs.modprobe_path ["gim"]);
+  (* Update the gpus even if the module was present already, in case it was
+   * already loaded before xapi was (re)started. *)
+  Xapi_pci.update_pcis ~__context ~host:localhost;
```

Comment:
Should this call be wrapped in `switched_sync`, the way we do it in `dbsync_slave.ml`?